File: src/main/java/com/almox/model/enums/FiltroConsideracaoAtivos.java
package com.almox.model.enums;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;

@Getter
@JsonFormat(shape = JsonFormat.Shape.OBJECT)
public enum FiltroConsideracaoAtivos implements IEnum {
    CONSIDERAR_TODOS("Considerar Todos"),
    APENAS_ATIVOS("Considerar Apenas Ativos"),
    APENAS_EXCLUIDOS("Considerar Apenas Excluídos");

    private String descricao;

    FiltroConsideracaoAtivos(String descricao) {
        this.descricao = descricao;
    }

    @JsonCreator
    public static FiltroConsideracaoAtivos deserialize(@JsonProperty("type") String type) {
        return IEnum.fromType(values(), type);
    }

    @Override
    public String getType() {
        return name();
    }
}

File: frontend/src/app/model/enums/filtro-consideracao-ativos.ts
export const FiltroConsideracaoAtivos = Object.freeze({
    TODOS: {
        type: "CONSIDERAR_TODOS",
        descricao: "Considerar Todos",
        valor: null,
    },
    APENAS_ATIVOS: {
        type: "APENAS_ATIVOS",
        descricao: "Considerar Apenas Ativos",
        valor: true,
    },
    APENAS_EXCLUIDOS: {
        type: "APENAS_EXCLUIDOS",
        descricao: "Considerar Apenas Excluídos",
        valor: false,
    },
    resolverSelecaoConsideracao(consideracao: any) {
        if (consideracao == null) return FiltroConsideracaoAtivos.TODOS;
        if (!!consideracao) return FiltroConsideracaoAtivos.APENAS_ATIVOS;
        if (!consideracao) return FiltroConsideracaoAtivos.APENAS_EXCLUIDOS;
        return undefined;
    },
});
